File: Homework/03.C# OOP/01.DefiningClassesPart1/DefiningClassesPart1/Display.cs

namespace DefiningClassesPart1
{
    using System;

    public class Display
    {
        private double? size;
        private int? numberOfColors;

        public Display()
        {
        }
        public Display(double? size = null,
                        int? numberOfColors = null)
        {
            this.Size = size;
            this.NumberOfColors = numberOfColors;
        }
        public double? Size
        {
            get { return this.size; }
            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("Size should be > 0");
                }
                else
                {
                    this.size = value;
                }
            }
        }
        public int? NumberOfColors
        {
            get { return this.numberOfColors; }
            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("Number of colors should be > 0");
                }
                else
                {
                    this.numberOfColors = value;
                }
            }
        }
        public override string ToString()
        {
            return String.Format("Size: {0}, Number of colors: {1}",
                                  this.Size, this.NumberOfColors);
        }
    }
}

File: Mobile Applications for iOS/phones/phones/Display.m
//
//  Display.m
//  phones
//
//  Created by z on 1/28/16.
//  Copyright © 2016 z. All rights reserved.
//

#import "Display.h"

@implementation Display

-(instancetype)initWithSize:(double)size andNumberOfColors:(int)numberOfColors{
    self = [super init];
    if (self) {
        self.size = size;
        self.numberOfColors=numberOfColors;
    }
    
    return self;
}

+(Display*)displayWithSize:(double)size andNumberOfColors:(int)numberOfColors{
    return [[Display alloc] initWithSize:size andNumberOfColors:numberOfColors];
}

-(NSString *)description{
    return [NSString stringWithFormat:@"Display Size: %f, Number of colors: %d",
            self.size,
            self.numberOfColors];
}


@end
